include/exadg/convection_diffusion/user_interface/application_base.h

```diff
@@ -92,15 +91,15 @@ class ApplicationBase
     param.check();
     param.print(pcout, "List of parameters:");
 
-    // grid
+    // create grid and mapping owned by driver
     GridUtilities::create_mapping(mapping, param.grid.element_type, param.mapping_degree);
-    create_grid();
-    print_grid_info(pcout, *grid);
+    create_grid(grid, mapping);
```

Comment:
before this line, `grid = std::make_shared<Grid<dim>>();`

Reply:
One could argue to shift this line even inside `create_grid()` to be consistent in terms of the `create()`-nomenclature, but then every concrete application has to write this line. For this reason, I would probably suggest to do it just once in `Application::setup()`.

What do you think?

Reply:
One example where we do it like this would be `Application::create_postprocessor()`, where the object is created inside the function. In this sense, it would be consequent to shift `grid = std::make_shared<Grid<dim>>();` inside `create_grid()`.

Reply:
I also favor keeping `grid = std::make_shared<Grid<dim>>();` outside of `create_grid` just to not repeat it all of the `application.h` even though this is a bit "inconsistent" given the function's name.